### PR TITLE
RFE #1020 Aero Maneuvers Tooltip Request

### PR DESCRIPTION
**Pull request by HammerGS**
Implements RFE #1020 to add informative tooltips to the aerospace maneuver selection dialog. Players can now hover over maneuver options to see detailed information about requirements, thrust costs, control modifiers, and effects.

Changes

Core Implementation:
- Added createManeuverTooltip() method to generate HTML-formatted tooltips for all 9 aerospace maneuvers
- Added getManeuverResourceKey() helper method for consistent resource string lookup
- Updated checkPerformability() to dynamically set tooltips based on current game state
- Tooltips display current values (velocity, altitude, thrust) when maneuvers are unavailable, helping players understand why they can't perform certain maneuvers

Internationalization:
- Added 47 new resource strings in messages.properties (English)
- Added full translations in messages_de.properties (German)
- Added full translations in messages_es.properties (Spanish)
- Added full translations in messages_ru.properties (Russian)
- All dynamic values use MessageFormat for proper localization

Testing:
- Created ManeuverChoiceDialogTest.java with 14 comprehensive unit tests
- Tests cover dialog initialization, resource string validation, maneuver requirements, thrust costs, control modifiers, and state management
- Configured for headless CI execution with @BeforeAll setup

Visual Design:
- Tooltips use color coding: violet headers, red text for unavailable conditions
- Clean HTML formatting with proper spacing and structure
- Dynamic values highlighted in red when requirements aren't met

Files Modified
- ManeuverChoiceDialog.java - Added tooltip generation and dynamic updates (+143 lines)
- messages.properties - English tooltip strings (+47 entries)
- messages_de.properties - German translations (+50 lines)
- messages_es.properties - Spanish translations (+46 lines)
- messages_ru.properties - Russian translations (+46 lines)
- ManeuverChoiceDialogTest.java - Unit tests (+368 lines, NEW FILE)

Total Changes: 6 files changed, 696 insertions(+), 4 deletions(-)

Test Plan
- All 14 unit tests pass locally
- Tests configured for headless CI execution
- Resource strings verified for all 4 languages
- Maneuver requirements logic validated
- Dynamic tooltip updates verified via state management tests

Notes
- Tooltips are fully localized and follow MegaMek i18n standards
- Implementation follows MegaMek Java 17 coding standards (K&R braces, 4-space indentation)
- All public methods documented with JavaDoc
- Test coverage includes edge cases for velocity/altitude requirements and VSTOL-specific logic

### CODE REVIEW

**Comment by HammerGS** [comment]
In additional player testing with the Test Pilots

File: megamek/src/megamek/client/ui/dialogs/phaseDisplay/ManeuverChoiceDialog.java [hunk 1]
```diff
@@ -294,11 +302,136 @@ public int getChoice() {
...
+
+        // Effect
+        tooltip.append(Messages.getString("ManeuverChoiceDialog.effectLabel")).append(": ");
+        tooltip.append(Messages.getString("ManeuverChoiceDialog." + maneuverKey + ".effect"));
+
+        tooltip.append("</BODY></HTML>");
+        return tooltip.toString();
+    }
+
+    /**
```
Comment (SJuliez) [inline megamek/src/megamek/client/ui/dialogs/phaseDisplay/ManeuverChoiceDialog.java]: This could be moved into the ManeuverType class as a public static.

File: megamek/unittests/megamek/client/ui/dialogs/phaseDisplay/ManeuverChoiceDialogTest.java [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+    void setUp() {
+        mockFrame = new JFrame();
+        dialog = new ManeuverChoiceDialog(mockFrame, "Test Maneuver Selection");
+    }
+
+    /**
+     * Test that dialog is properly initialized with all maneuvers.
+     */
+    @Test
+    void testDialogInitialization() {
```
Comment (SJuliez) [inline megamek/unittests/megamek/client/ui/dialogs/phaseDisplay/ManeuverChoiceDialogTest.java:82]: I doubt that this makes sense. The title could be internationalized. Intentional changes to the title are a normal thing; I'd rather not have to update two places just to change a GUI text. And the title will not make the dialog fail on a player.

File: megamek/unittests/megamek/client/ui/dialogs/phaseDisplay/ManeuverChoiceDialogTest.java [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+        assertNotNull(dialog, "Dialog should be initialized");
+        assertEquals("Test Maneuver Selection", dialog.getTitle(),
+            "Dialog title should match");
+    }
+
+    /**
+     * Test that all maneuver resource keys exist in messages.properties.
+     */
+    @Test
+    void testAllManeuverResourceKeysExist() {
```
Comment (SJuliez) [inline megamek/unittests/megamek/client/ui/dialogs/phaseDisplay/ManeuverChoiceDialogTest.java]: This can make sense on the ManeuverType class but not for the dialog.

File: megamek/unittests/megamek/client/ui/dialogs/phaseDisplay/ManeuverChoiceDialogTest.java [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+            "Current values (plural) format should exist");
+        assertResourceExists("ManeuverChoiceDialog.notVSTOL",
+            "Not VSTOL message should exist");
+    }
+
+    /**
+     * Test that None maneuver is always available.
+     */
+    @Test
+    void testNoneManeuverAlwaysAvailable() {
```
Comment (SJuliez) [inline megamek/unittests/megamek/client/ui/dialogs/phaseDisplay/ManeuverChoiceDialogTest.java:128]: This also doesn't relate to the dialog, more to maneuvers themselves.

File: megamek/unittests/megamek/client/ui/dialogs/phaseDisplay/ManeuverChoiceDialogTest.java [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+            "VIFF at velocity 5 should cost 7 thrust (velocity + 2)");
+        assertEquals(12, ManeuverType.getCost(ManeuverType.MAN_VIFF, 10),
+            "VIFF at velocity 10 should cost 12 thrust (velocity + 2)");
+    }
+
+    /**
+     * Test maneuver control roll modifiers.
+     */
+    @Test
+    void testManeuverControlModifiers() {
```
Comment (SJuliez) [inline megamek/unittests/megamek/client/ui/dialogs/phaseDisplay/ManeuverChoiceDialogTest.java]: I wouldn't write tests that nail down rule numbers. Changes will then require two updates and those numbers will not crash MM for a player, at worst they're just incorrect until someone notices.
